numpy/_core/fromnumeric.py

```diff
@@ -296,7 +303,16 @@ def reshape(a, newshape, order='C'):
            [3, 4],
            [5, 6]])
     """
-    return _wrapfunc(a, 'reshape', newshape, order=order)
+    if newshape is None and shape is None:
+        raise TypeError(
+            "reshape() missing 1 required positional argument: 'shape'")
+    if newshape is not None and shape is not None:
+        raise ValueError(
+            "You cannot specify 'newshape' and 'shape' arguments "
+            "at the same time.")
+    if shape is not None:
+        newshape = shape
```

Comment:
as alluded to above, instead of this let's make `newshape` keyword 
-only and also issue a deprecation warning if anyone passes it. Old uses with a positional argument continue to work with the new argument name. Old uses with a keyword argument continue to work, but with a deprecation warning so they can fix their code when we finally remove newhape.

Please also mark here in a comment that the deprecation happened in numpy 2.1.

Reply:
Done!